### PR TITLE
add cors helper

### PR DESCRIPTION
**Pull request by ianmacartney**
a sketch of what it could look like w.r.t. composition vs. subclassing

----

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by ianmacartney** [comment]
```ts
const http = httpRouter();

const httpWithCORS = corsRouter(http, {
  allowedOrigins: ["*"], // allow regex
  allowCredentials: false, // if true, must specify non-* origins
  allowMethods: ["*"],
  allowHeaders: ["*"],
  exposeHeaders: [],
  maxBrowserCacheAge: 600,
});

// route with the one based on what you want.
httpWithCORS.route({
  allowMethods: // overrides
  // normal fields
  path: "/myfn",
  action: "POST",
  handler: myhandler,
});
```

File: packages/convex-helpers/server/cors.test.http.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+cors.route({
+  path: "/fact",
+  method: "DELETE",
+  handler: everythingHandler,
+});
+
+/**
+ * Non-CORS routes
+ */
+http.route({
```
Comment (sshader) [inline packages/convex-helpers/server/cors.test.http.ts:52]: Can you do things like route to `/foo` with method GET with CORS and `/foo` with method POST without CORS? From a quick scan, looks like yes. I guess the OPTIONS will only reflect the ones with the CORS router, which is fine, but also probably not a super common case

**Review by sshader — APPROVED** [review]

File: packages/convex-helpers/server/cors.test.http.ts (hunk 1 above)
Comment (ianmacartney) [reply packages/convex-helpers/server/cors.test.http.ts:52]: Actually that's a bit of a rough edge: I believe if you route w/o CORS, then with CORS, the CORS router will include the non-cors method in the accepted methods

Comment (ianmacartney) [reply packages/convex-helpers/server/cors.test.http.ts:52]: I'll file an issue to track it

**Pull request merged**